cmd/mosn/main/mosn.go

```diff
@@ -74,7 +74,7 @@ import (
 	_ "mosn.io/mosn/pkg/upstream/servicediscovery/dubbod"
 	_ "mosn.io/mosn/pkg/wasm/abi/proxywasm010"
 	_ "mosn.io/mosn/pkg/wasm/abi/proxywasm020"
-	_ "mosn.io/mosn/pkg/wasm/runtime/wasmer"
+	_ "mosn.io/mosn/pkg/wasm/runtime/wazero"
```

Comment:
keep both wazero and wasmer, so that users can choose which one to use

Reply:
sure but this will keep mosn main portability coupled to what libraries wasmer distributes. can do

Reply:
what if we keep import both vms, and add a makefile target like `build-local-wasmer` which build in wamser, while the default one `build-local` just build with wazero?

Reply:
so let's clarify, as I think really there are a few things. also keep in mind wasmer 3 just came out in rust and once goes to wasmer-go likely will be incompatible in some ways.

right now build-local adds wazero because it is not tag-specific. Users can already supply TAGS to the makefile to add wasmer, which will result in wazero and wasmer as engine choices. So, basically build-local-wasmer is same as setting TAGS=wasmer

If the desire is to intentionally remove wazero (2MB from the 140+MB binary), then we have to add a build tag to disable wazero. However, doing that forces us to make tests depend only on mock and basically that ensures they won't happen. Also, it provides no out-of-box wasm runtime w/o CGO which I thought was the original intent of the request to add wazero. https://github.com/mosn/proxy-wasm-go-host/issues/15

thoughts?